ticker/src/main/java/com/robinhood/ticker/TickerColumnManager.java

```diff
@@ -36,33 +39,48 @@
     final ArrayList<TickerColumn> tickerColumns = new ArrayList<>();
     private final TickerDrawMetrics metrics;
 
-    // The character list that dictates how to transition from one character to another.
-    private char[] characterList;
+    private List<char[]> characterLists;
     // A minor optimization so that we can cache the indices of each character.
-    private Map<Character, Integer> characterIndicesMap;
+    private List<Map<Character, Integer>> characterIndicesMaps;
+    private Set<Character> supportedCharacters;
 
     TickerColumnManager(TickerDrawMetrics metrics) {
         this.metrics = metrics;
     }
 
     /**
-     * @see {@link TickerView#setCharacterList(char[])}.
+     * @inheritDoc TickerView#setCharacterLists
      */
-    void setCharacterList(char[] characterList) {
-        this.characterList = characterList;
-        this.characterIndicesMap = new HashMap<>(characterList.length);
+    void setCharacterLists(char[]... characterLists) {
+        this.characterLists = new ArrayList<>(characterLists.length);
+        this.supportedCharacters = new HashSet<>();
+        for (char[] characterList : characterLists) {
+            final char[] modifiedCharacterList = new char[characterList.length + 1];
+            modifiedCharacterList[0] = TickerUtils.EMPTY_CHAR;
+            System.arraycopy(characterList, 0, modifiedCharacterList, 1, characterList.length);
+            this.characterLists.add(modifiedCharacterList);
+
+            for (char c : modifiedCharacterList) {
+                supportedCharacters.add(c);
+            }
+        }
 
-        for (int i = 0; i < characterList.length; i++) {
-            characterIndicesMap.put(characterList[i], i);
+        this.characterIndicesMaps = new ArrayList<>(this.characterLists.size());
+        for (char[] characterList : this.characterLists) {
```

Comment:
change to int based for loop

Reply:
no performance penalties for these because `setCharacterList` will only be called once on initialization

Reply:
I'd still change it, tbh. It's a lot easier to be zero tolerance on these for-loops than try to guess the impact of each one.